src/context/campaignConfigsStore.tsx

```diff
@@ -107,27 +134,80 @@ export const CampaignConfigsStoreContextProvider: FunctionComponent = ({
     setAllConfigs({});
   }, []);
 
-  const [, setState] = useState();
   useEffect(() => {
-    const loadCampaignConfigFromStore = async (): Promise<void> => {
-      const campaignConfigsString = await AsyncStorage.getItem(
-        CAMPAIGN_CONFIGS_STORE_KEY
-      );
-      if (!campaignConfigsString) {
-        setHasLoadedFromStore(true);
-        return;
+    /**
+     * Migrates credentials from old config store to new config store. Clears all old storage locations.
+     * @param newStorageHasData whether most recent data has been found. If this is true, does not
+     * attempt any migration and just clears old storage locations
+     * @returns the updated value of {@link newStorageHasData}, i.e. true if any updated data was found
+     * in the older storage (if {@link newStorageHasData} was passed in as true, it will always return true)
+     */
+    const migrateOldConfigFromStore = async (
+      newStorageHasData: boolean
+    ): Promise<boolean> => {
+      let hasUpdatedData = newStorageHasData;
+      // check v1 storage
+      if (!hasUpdatedData) {
+        const configString = await AsyncStorage.getItem(
+          CAMPAIGN_CONFIGS_STORE_KEY
+        );
+        if (configString) {
+          try {
+            const configStringFromStore: CampaignConfigsMap = JSON.parse(
+              configString
+            );
+            setAllConfigs(configStringFromStore);
+            hasUpdatedData = true;
```

Comment:
since i notice that `hasUpdatedData` will only remain false if this `try catch` fail.

so maybe we can put the `Sentry` right after the catch and right before `return hasUpdatedData`

wdyt?

Reply:
`hasUpdatedData` is not guaranteed to be `true` at the end of `migrateOldConfigFromStore` even if there is no error because there is a chance that AsyncStorage does not have any data as well (`configString === null`), in which case, there is nothing to migrate and the function will return false